weld/exprs.rs

```diff
@@ -18,20 +18,21 @@ pub fn new_expr(kind: ExprKind<Type>, ty: Type) -> WeldResult<Expr<Type>> {
 }
 
 pub fn literal_expr(kind: LiteralKind) -> WeldResult<Expr<Type>> {
-    new_expr(Literal(kind),
-             Scalar(match kind {
-                        BoolLiteral(_) => ScalarKind::Bool,
-                        I8Literal(_) => ScalarKind::I8,
-                        I16Literal(_) => ScalarKind::I16,
-                        I32Literal(_) => ScalarKind::I32,
-                        I64Literal(_) => ScalarKind::I64,
-                        U8Literal(_) => ScalarKind::U8,
-                        U16Literal(_) => ScalarKind::U16,
-                        U32Literal(_) => ScalarKind::U32,
-                        U64Literal(_) => ScalarKind::U64,
-                        F32Literal(_) => ScalarKind::F32,
-                        F64Literal(_) => ScalarKind::F64,
-             }))
+    new_expr(Literal(kind.clone()),
+             match kind {
+                 BoolLiteral(_) => Scalar(ScalarKind::Bool),
+                 I8Literal(_) => Scalar(ScalarKind::I8),
+                 I16Literal(_) => Scalar(ScalarKind::I16),
+                 I32Literal(_) => Scalar(ScalarKind::I32),
+                 I64Literal(_) => Scalar(ScalarKind::I64),
+                 U8Literal(_) => Scalar(ScalarKind::U8),
+                 U16Literal(_) => Scalar(ScalarKind::U16),
+                 U32Literal(_) => Scalar(ScalarKind::U32),
+                 U64Literal(_) => Scalar(ScalarKind::U64),
+                 F32Literal(_) => Scalar(ScalarKind::F32),
+                 F64Literal(_) => Scalar(ScalarKind::F64),
+                 StringLiteral(_) => Vector(Box::new(Scalar(ScalarKind::U8))),
```

Comment:
In type inference below this pushed type is `I8` (instead of `U8`), and in the test above it's cast into a `u8`. Make sure it's consistent (I'd make it `i8` everywhere, so change this and the test above to cast to `i8`).